packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -1133,24 +1139,28 @@ abstract class _RenderCustomClip<T> extends RenderProxyBox {
   }
 
   void _markNeedsClip() {
-    _clip = null;
+    _clipShape = null;
     markNeedsPaint();
     markNeedsSemanticsUpdate();
   }
 
-  T get _defaultClip;
-  T _clip;
+  T get _defaultClipShape;
+  T _clipShape;
+
+  final Clip initialClip; // May be overridden by clipper.
```

Comment:
i don't understand the purpose of initialClip

Reply:
Renamed it to defaultClip. Its purpose is to serve when the clipper is null: `Clip get clip => clipper == null ? defaultClip : clipper.clip;`

Reply:
ah. add docs explaining all this. :-)

Reply:
also, this should be mutable, otherwise you won't be able to change it when you update the render object widget.

Reply:
maybe it should just be called the same as all the other properties that do the same thing?

Reply:
Done. BTW, it seems that our current API doesn't have any way of modifying `_clipBehavior` (the old `_defaultClipBehavior` and `initalClip`) when the render object widget is updated? The `_clip` is updated according to size, but `_clipBehavior` seems to remain constant with different sizes.